src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,97 @@
+import os
+import tempfile
+
+import unittest
+import unittest.mock
+
+import insights_client
+
+
+class TestMOTD(unittest.TestCase):
+    """Verify the behavior of MOTD file."""
+
+    def setUp(self):
+        self.dir = tempfile.TemporaryDirectory()
+
+        # The source file in /etc/insights-client/
+        self.dir_src = unittest.mock.patch(
+            "insights_client.MOTD_SRC", f"{self.dir.name}/src"
+        )
+        self.dir_src.start()
+        # Actual file in /etc/motd.d/
+        self.dir_motd = unittest.mock.patch(
+            "insights_client.MOTD_FILE", f"{self.dir.name}/motd"
+        )
+        self.dir_motd.start()
+        # /etc/insights-client/.registered
+        self.dir_reg = unittest.mock.patch(
+            "insights_client.REGISTERED_FILE", f"{self.dir.name}/reg"
+        )
+        self.dir_reg.start()
+        # /etc/insights-client/.unregistered
+        self.dir_unreg = unittest.mock.patch(
+            "insights_client.UNREGISTERED_FILE", f"{self.dir.name}/unreg"
+        )
+        self.dir_unreg.start()
+
+        with open(f"{self.dir.name}/src", "w") as f:
+            f.write("This is the content of the MOTD file.")
+
+    def tearDown(self):
+        self.dir_unreg.stop()
+        self.dir_reg.stop()
+        self.dir_motd.stop()
+        self.dir_src.stop()
+        self.dir = None
+
+    def test_present(self):
+        assert not os.path.exists(insights_client.MOTD_FILE)
+
+        # The file gets created/symlinked when .registered & .unregistered do not exist
+        insights_client.update_motd_message()
+        assert os.path.exists(insights_client.MOTD_FILE)
+        assert os.path.samefile(insights_client.MOTD_FILE, insights_client.MOTD_SRC)
+
+    def test_absent_on_dot_registered(self):
+        # The MOTD file exists by default
+        insights_client.update_motd_message()
+        assert os.path.exists(insights_client.MOTD_FILE)
+
+        # It gets removed when .registered or .unregistered exists
+        with open(f"{self.dir.name}/reg", "w") as f:
+            f.write("")
```

Comment:
if all you want to do here is create an empty file, then you can do it in a simpler way (IMHO):
```suggestion
        os.truncate(f"{self.dir.name}/reg", 0)
```
(the same applies to the other occurrences of this code)

Reply:
`truncate` operates on existing files, this raises `FileNotFoundError`s.